mreg_cli/api/models.py

```diff
@@ -126,6 +131,38 @@ def parse(cls, value: Any, mode: IPNetMode | None = None) -> IP_AddressT | IP_Ne
             return func(ipnet)
         return ipnet.ip_or_network
 
+    @classmethod
+    def parse_ip_optional(
+        cls, ip: str, version: Literal[4, 6] | None = None
+    ) -> IP_AddressT | None:
+        """Check if a value is a valid IP address.
+
+        :param ip: The IP address to parse.
+        :param version: The IP version to parse as. Parses as any version if None.
+        :returns: The parsed IP address or None if parsing fails.
+        """
+        mode = "ipv4" if version == 4 else "ipv6" if version == 6 else "ip"
+        try:
+            return cls.parse(ip, mode=mode)
+        except ValueError:
+            return None
+
+    @classmethod
+    def parse_network_optional(
+        cls, ip: str, version: Literal[4, 6] | None = None
+    ) -> IP_NetworkT | None:
```

Comment:
Added to match `parse_ip_optional`, but it's unused (as of now).  Adding unused stuff sucks, but if we want to use `parse_ip_optional`, it's likely we want to use `parse_network_optional` somewhere too.